animation_nodes/nodes/mesh/set_vertex_weights.py

```diff
@@ -0,0 +1,131 @@
+            self.newInput("Integer", "Group Index", "groupIndex")
+        elif self.groupIdentifierType == "NAME":
+            self.newInput("Text", "Name", "groupName")
+
+        if self.mode == "INDEX":
+            self.newInput(VectorizedSocket("Integer", "useIndexList",
+                ("Index", "index"), ("Indices", "indices")))
+        self.newInput(VectorizedSocket("Float", "useWeightsList",
+                ("Weight", "weight"), ("Weights", "weights")))
+        
+        self.newOutput("Object", "Object", "object")
+
+    def draw(self, layout):
+        layout.prop(self, "mode", text = "")
+
+    def drawAdvanced(self, layout):
+        layout.prop(self, "groupIdentifierType", text = "Type")
+
+    def getExecutionFunctionName(self):
+        if self.mode == "INDEX":
+            if self.useIndexList:
+                if self.useWeightsList:
+                    return "execute_IndicesList"
+                else:
+                    return "execute_IndicesSingle"
+            else:
+                return "execute_Index"
+        elif self.mode == "ALL":
+            if self.useWeightsList:
+                return "execute_AllList"
+            else:
+                return "execute_AllSingle"
+
+    def execute_Index(self, object, identifier, index, weight):
+        if object is None: return
+        vertexGroup = self.check(object, identifier)
+
+        vertexGroup.add([index], weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_IndicesSingle(self, object, identifier, indices, weight):
+        if object is None: return
+        vertexGroup = self.check(object, identifier)
+
+        vertexGroup.add(indices, weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_IndicesList(self, object, identifier, indices, weights):
+        if object is None: return
+        vertexGroup = self.check(object, identifier)
+
+        weights = VirtualDoubleList.create(weights, 0).materialize(len(indices))
+        for i, index in enumerate(indices):
+            vertexGroup.add([index], weights[i], "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_AllSingle(self, object, identifier, weight):
+        if object is None: return
+        vertexGroup = self.check(object, identifier)
+
+        indices = list(range(len(object.data.vertices)))
+
+        vertexGroup.add(indices, weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_AllList(self, object, identifier, weights):
+        if object is None: return
+        vertexGroup = self.check(object, identifier)
+
+        weights = VirtualDoubleList.create(weights, 0).materialize(len(object.data.vertices))
+        for i, weight in enumerate(weights):
+            vertexGroup.add([i], weight, "REPLACE") 
+        object.data.update()
+        return object            
+
+    def check(self, object, identifier):
```

Comment:
The function name is not intuitive. The function validates the inputs, throws exceptions, and returns the vertex group. So perhaps the code should be moved to the `getVertexGroup` function.

Reply:
I have moved the check-conditions into the `getVertexGroup`.